evm_arithmetization/tests/log_opcode.rs

```diff
@@ -469,12 +469,25 @@ fn test_log_with_aggreg() -> anyhow::Result<()> {
         &all_stark,
         &config,
         inputs_first,
-        max_cpu_len_log,
+        // We want only one segment.
+        20,
         &mut timing,
         None,
     )?;
 
-    assert_eq!(segment_proofs_data_first.len(), 2);
+    assert_eq!(segment_proofs_data_first.len(), 2); // second one is a dummy segment
+
+    let (segment_agg_proof_first, updated_agg_public_values_first) = all_circuits
+        .prove_segment_aggregation(
+            false,
+            &segment_proofs_data_first[0].proof_with_pis,
+            segment_proofs_data_first[0].public_values.clone(),
+            false,
+            segment_proofs_data_first[1].is_dummy,
+            &segment_proofs_data_first[1].proof_with_pis,
+            segment_proofs_data_first[1].public_values.clone(),
+        )?;
```

Comment:
Can't we have `prove_segment_aggregation` take only 4 arguments, i.e.
- `lhs_is_agg`
- `lhs_segment_data`
- `rhs_is_agg`
- `rhs_segment_data`
?

This would require an API change on `zero-bin` to accomodate, but we're already doing a bunch of breaking changes with continuations. Seems it would be cleaner to handle that way.

Reply:
I don't think it even would, since the API function used by `zero-bin` is actually
```rust
pub fn generate_segment_agg_proof(
    p_state: &ProverState,
    lhs_child: &SegmentAggregatableProof,
    rhs_child: &SegmentAggregatableProof,
    has_dummy: bool,
) -> ProofGenResult<GeneratedSegmentAggProof>
```
which can be left as is.

Reply:
Technically not part of this PR but related: the [description](https://github.com/0xPolygonZero/zk_evm/blob/6eee424ce3430e92d85db729fa1bedd708fa15b5/evm_arithmetization/src/fixed_recursive_verifier.rs#L1714) of `prove_segment_aggregation` is out of date.